source/boundary_temperature/function.cc

```diff
@@ -40,7 +42,28 @@ namespace aspect
     boundary_temperature (const types::boundary_id /*boundary_indicator*/,
                           const Point<dim> &position) const
     {
-      return boundary_temperature_function.value(position);
+      if (coordinate_system == ::aspect::Utilities::Coordinates::CoordinateSystem::cartesian)
+        {
+          return boundary_temperature_function.value(position);
+        }
+      else if (coordinate_system == ::aspect::Utilities::Coordinates::CoordinateSystem::spherical)
+        {
+          const std_cxx11::array<double,dim> spherical_coordinates =
+            aspect::Utilities::Coordinates::cartesian_to_spherical_coordinates(position);
+          Point<dim> point;
+
+          for (unsigned int i = 0; i<dim; ++i)
+            point[i] = spherical_coordinates[i];
+
+          return boundary_temperature_function.value(point);
+        }
+      else
+        {
+          const double depth = this->get_geometry_model().depth(position);
+          Point<dim> point;
+          point(0) = depth;
+          return boundary_temperature_function.value(point);
+        }
```

Comment:
add
```
else
AssertThrow(false, ExcNotImplemented());
```
here and in all other cases. The reasoning is that if the enum changes at any time in the future, it won't silently generate wrong results.